images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -87,19 +96,37 @@ function Install-JavaFromAdoptOpenJDK {
     New-Item -ItemType File -Path $javaVersionPath -Name "$Architecture.complete" | Out-Null
 }
 
-$jdkVersions = (Get-ToolsetContent).java.versions
+$jdkVendors = (Get-ToolsetContent).java.vendors.name
+$defaultVendor = (Get-ToolsetContent).java.default_vendor
 $defaultVersion = (Get-ToolsetContent).java.default
 
-foreach ($jdkVersion in $jdkVersions) {
-    Install-JavaFromAdoptOpenJDK -JDKVersion $jdkVersion
+# We get the difference between Adopt and Adoptium and then assign it as Adopt's JAVA_HOME
+$uniqVersion = (Get-ToolsetContent).java.vendors.versions | Group-Object -noelement | Where-Object {$_.count -eq 1} | Select-Object -expand Name
+
+foreach ($jdkVendor in $jdkVendors) {
```

Comment:
Just a proposal for discussion:
May be it doesn't make sense to iterate through the vendors here.
We can create two loops:
```
foreach ($jdkVersion in $temurinVersions) {
  Install-JavaJDK -Version $jdkVersion -FullVendorName "Temurin"
  Set-JavaPath -Version $jdkVersion
}

foreach ($jdkVersion in $adoptVersions) {
  Install-JavaJDK -Version $jdkVersion -FullVendorName "Adopt"
}
```

Your current approach is more flexible but we won't have more vendors in future. It is 100%. Having two vendors is temporary solution.

With two looks we can just get rid of logic with default_vendor and simplify a bit.